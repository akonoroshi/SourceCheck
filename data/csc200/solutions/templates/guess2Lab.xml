<project name="guess1Lab" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1" guid="541caacc-4c0e-4d35-b39a-9b0babafceb9" assignment="none"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAARIUlEQVR4Xu2dB1SUxxbH7+6yu3QRBaTb2wt2sXexd8Fu1GieJclRo7FGY6+JJvGY8+Kzo8RnIjGaWBBFRBBEURQBu0F9R41PiRWkvXMHd9lhd3G/leX7Vu+c4zHZnZk789+f830zc+eOrKCgoAAokQIiKSAjAEVSnswyBQhAAkFUBQwCiE/lgwcPwuXLl0VtnMZ4o0aNoF27dpJoCzWidBXQA/DChQvg4OBQqlb8/fwgdKAX2CuFVdttcTg8VHtpCzk7O4Obm5uwSii3pBXgAMSRzxKjXtyM1jCojr1ZQtwZE6Etl5+fD3Xr1jWrHiokTQU4AENDQ6FBgwaCWvrkyRPAkamklLqgI/SsbqeXRR28CpRNBsOzWf5Gi98c8Qf3XUBAgKD2UWZpK8ABOHfuXBg4cKC2xY8fPwZPT08YPHgwzJw5k40+aWlpsHr1amjRogUsXryYjZhjx46FL7/8ErKzs6F///4wfPhwOHz4MJw5c4bVdXVpV4MAOi5KBVA7wrOZfkZVujJ4L/cdvg9SencU4ABctGgRdO3aVdu7zMxMaN68OZQvXx6+++47BuKjR48YiDgpWLFiBeTm5sKxY8cgMDAQ1Go1ICCYd9euXbB582ZW161VvQwCiN/ZNBoIuUl7jCp6acDP3HfNmjV7d9SnnvDLMBcvXgR8pOqmpk2bwpUrV+CDDz6Av//+mz1uMU9KSgq0atWKZY2NjWX/jX9jPsxz584dyMjIYN/fWdPPIIB2H4eBonrrEkfA5D4/ce3R2KTf7t1QQG8W/MMPP5T6i379CgWg3DHKDMVkcKb7Nm25mJgYmDdvnhn1UBGpKmBwHfCLL76ALl26SKrNkZGRsHLlSkm1iRrz9goYBPDu3buQmpoKz58/f3sLb6gBl34qVqwI+G6nUqksbo8MSEsBPQBxB6RGjRqCWvnkZCjcO/wj2NqYXkwpl0GztWch49ELVujly5dQpUoVcHR0NL0Symn1CnAAvnr1Cq5fvy64U5mrO0GAmwD6NBZkMrg96pDWHs6waZIhWH6rLsABuHXrVmjcuLHRDuE2Xffu3QEf0brJ4cfu4O4g1ytn03AAyN0qw6uINYbrlMng5vDftd/h47hevXpWLSg1XpgCJS5Enzt3DgYNGgSfffYZ7N+/H6ZOnQoTJkyApKQkGDduHOzdW7hI7LK5j0EA5X4NwaZ2hxIBvDLoV67FtNAs7Ae09twcgLiw3KFDB22fzp8/D0OGDIHvv/+eLSyHhITApEmTtIvRp06dYnndQgcaBBBABqouU0sE8FL/3ZyGtNBs7UgJaz8HIC4e37p1i6vBy8sL7OzsALfl8B3N3d0dnJycICsrS/so9t41BNwMPIIRQJu6nSE39YjRR/D5Xju13+GuSvv27YX1gHJbtQJ6s+A1a9ZAw4YNBXWq6a1QgPRIQWUwc97oMEi6l8O9A3bs2FFwPVTAehUwuA64fft2UCgUZbYkcu/ePahduzY5nVovR2a33CCA6MWC63L4qC2rhBDi6EeL0WWluDTs6AG4ceNGs0ai9G9HgvvLDLBXykzumUIGUGNeBNx6UbgDgg6n6PHs6upqch2U0boV4AB8+vQp3L5926weZa8NghquCuFlZTLI+PCgttyLFy+gSZMmwuuhElapAAfghg0bmP+fOclxQw9uKUbVaTLkXo4Cm1rtQdVuPBSADJ7PN+BOL5PBjWH7OZO0GG3OL2CdZfQWogcMGKDtCa4DonfzV199BXv27IH169fDrFmz4OjRo2y9ENcGccTCVH5LX721QGXzEaCo2hzklWqBzLYcPF8WqK+STAaXQ8K5z0vajbFOmanVxhTgAMQFZ/Rs1qTk5GRuIRrfzUaNGsUWpDt37sycT8ePH8+yu+8I5gCU+9QH+0nh8GxONXCYdQpkto7wbIGB8xwyGaT0+w/XPnNHYfqZrU8BDkAczc6ePcv1AhehcUkGHRXwj663Sl5eHpstY/LdPRQq2hftB8s9agIobQFysgrry8uF/Ic3DI6A53ru0H6OXte9evWyPiWpxWYpoDcLRqdPdMMXmlrY3YO8vbOEFoO8KdFwNu2mthwuguMZFErvhwIG1wHxbAi+5+G2W1klfN+cPXs2wVdWgkvEjkEAT5w4wQ4WSXU9DpeL0tPT2RFQuVzfDUwi2lIzTFBAD8Bt27ZZjUPA/fv3uUmTCf2lLBJTgAMQj1uix4tuwuOXmomHn58f261Ajxk8sJ6Tk8OOX+JsGPPheeGHDx+yz729vdln+J0lE56UGzlypCVNUN0WVIADcN26ddwMNC4ujh1Gx1kpzoBxmw4fzfv27WOH0j08PGDy5MksSgLu48bHx7MoCZgHHVnDw8MB68R9Xksl/AfQpk0bS1VP9VpYAQ5ADLmhG5oDR0T0A8SZca1atRhkeGZj+fLlDC6lUskmDghodHQ0O0mHURIwjR49mr1D+vv7W9SpAUds2rqzMCUWrJ4DcNWqVRyAOLrg3jAez8SFZykmBNCcZSMp9uV9bJOeS77uCIiCqFUqyM7OAilEki4oyAe5gj99hwCSG7/1ossBiI9WXQB9fXwgYkxlqOeuAnSdypbbwVNFyaHYSlsKp7wnoM4v3G2xtZGBcl4y/PXXQ60ZBJC27kpb9bKrjwNw2bJlHIDw6iV4hvYBuQzglyrToEP3ftqW4fpbbs4rQS1V2CjBnKD8UQf3QvDNb5gtxxV/wuUrV7R28R2VABT0M0gqMwfg0qVL9QD02dkXfvKYAEEDhmkbnpWWAFdXjQcPtenOp1jYWSkD23VJZgkQER4Gw+7/iwGYrhO7GgHEWTgl61SAA3DJkiUcgLKcLEAAH4w/znqHzgfomHBwcH0I8hQY8Pm1PjbfntGOglevXoWaNWuarJz7j+0ZgGnp6dwI2LJlS5ProIzSUoADECOe6r4DynOzGYD3/xkFwcHBcOTIERYj8PDQBtC5Eg9gte3pkHloG9jXbwNZV8/BXxvnGuypYm2i9nMEUEgcGo8NHRiAqWlpHIAUzkNaUAlpDQcgRkjVBVCR94oBeO/jwrU9FxcXwKipEcMa6gGI31cYNhtcgyfD3QUhYFPBE55E8YfOMY98zWmQyWRw7do1NhJWr17d5PZW+ndHBuCl1FRtGWwPAWiyhJLLyAG4cOFCDkCb/BwG4H/HFp75xYVlnHVGjmhsFEDndsHw+Lf1kHmgMDyvXvo6nj3GMeEIWLVqVZNF8drUmQGYcukSB2Dr1q1NroMySksBDsAFCxZwACoLchmAdz/iIxskjm8Dgc65ZvWkYPUp9i6JCRe5K1eubHI93puDGIAXU1I4AGkrzmQJJZeRAxDPfug+glWQxwC8Pfow13BvT0/IXNhfcGccQqbDg/Kmj3jFDfhu7coAvHDxIgdg27ZtBbeFCkhDAQ7A+fPncwCqZfkMwAydGH7Y7Co+XrCxky842QhbhukyZy1kVjN/ycRvWzcGYPKFCxyAdI2XNGAypxUcgBgAXHcEtJUXMAD/1Dm3i0YiPmwOPb3NW4bJWR4DDx48YG3FSQh61Jia/Ld3ZwCeT07mAKSARqYqKL18HIB42YwugHYKAO+dfeHWiD9YxAL0etmyZQvEjG1pcBLi93UEqPzrwrUQH6M9zV4aDfb2hdd24SQE/QZNTZV39ASnFX/CufPntUVwWYgANFVB6eXjAJwzZw4HoINSzgDURDHFvWJ0vzo5rpVBALF7VbekwI0xxp1Qs5Yc116GiABi+DdTU5WdvRiAZ5OKdlMQQIqoZaqC0svHAYhw6Y6AjioFA/D60H0wffp0wLvk0A0+7p9tjAKIa4H/C1tutKcvFh1j/oE+Pj4QFRUFlSpVMlmVaj/1YQCe0Tk6igB26tTJ5Dooo7QU4ADEqAe6ADqpbRiA14b8xhxSMeHdcAkT2kJHD/13QPeJ34BttXqQMT3IaC+fLzyqvdwQR0Ah74DVd/VlACa+voMOjaDTLAEoLaiEtIYDEC8k1AXQ2VbJACwex9lflQOP5vYQYoflVbj5QuYkIwvUJtRWc3d/BuDpxKLtPARQqs6yJnTpvc/CAThjxgwOwHJ2KgZg8dgtYqlW6+cBDMCE06e1TUAAg4KMj7hitZXsmqYAByBe0aU7ArrYqxmA6cHGb7M0zUzp5Kr9y0AGYHxCAgeg1K4VK53evh+1cADiREMXwPIOtgzA1GJXpoolTd3wEAbgqfh4DkDdK2bFahvZNU8BDsBp06ZxALo62jEAi1+lYJ6pty/1j18HMQDjXl8PoZmEdOvW7e0rpxpEUYAD8PPPP+cArOBkzwC82HeXKI0rbjTgtyEMwNi4OO1XGKbDEgDWqVMH0D0NQ9FRspwCHIB4E5LuI7iiswMD8EKxS6Mt15ySa663bygD8GRsLAcgXh9miYTBOjVnWNB/8cCBA+Dr62sJU+9tnRyAU6ZM4QB0K+fIAEzuHSYJgervH8YAjDl5kgOwRw/hS0KmdggDIBVP6FCL4Ufw+jJKb6cAByCG2dAdAd1dnBiAugEk386c6aV79uwJz5490xY4fvw4NDowkgF4IiaGAxDzWioZAxBDlfz+e9FFi5ay/67XywGIlxLqAljJtRwDMKlHqCg64B4vBkPCLTscdTQAHo+O1rYHIbVURFXdRzBeY4uPYNxCpFR6CnAAfvrppxyAnhVcGIBnum1jjxwM0YHnQgwlHBHedMkMumEVD3qJLv4IEUbeelNqcmgUGwGjjhee0sOEZXv37v2mooK/x5ub0PnC0AgouDIqYFQBDsBPPvmEA9CrYnkGYGLXrSzIECY8GWfoKCU6FWB8ac25EYw3jSfecL+3XLlyEBYWxgDFXQv8HEc2PNyOe8yRkZElvtxj8KHExEQIjBjDADwWFaXtEP6jsASAxEzZKMABiFex6j6Cvd1cmUNqfNBmdh0DOg5gWDYEtXhCtyo844EeMxkZGYATmoCAAPbIwndLHEFx2wydXhFAzVWvWA/WiY4QhhJG7T/9euut+ZGPmEPq0dcRuDA/AtinT5+yUYuslLoCHIATJ07kAPRxr8AAjGi8Euo3DmSj1JgxY1joteIJHUsjIiLYJdY4uiF0GDID403v3r2bgYLXseKF13iQPDY2lr3XYX1r1641+GjHoEMJCQks+NCRQ39A17MzGYCRR49yj+B+/YpChpS6QlShRRXgAMSrWnHU0iRfj4qFkRHyHGCTTW+o36iJ0XdAY63E9TMcOYVefIjQYcBLhDg2+hjUPjEH3BXPGYBHIouuhsWYhAg7JetUgAMQF13xHU+TXB1UUHvfh+x/s/IVEJ7XFJwDi25SKosuP038FfrLT4OtvPAop92ymyyCvyap1WqzLlcsi7aTjTcrwAGI2fG+ODyrq9kBaPlgHygeF93j8eYqLZcjz7UqxLkVzXjxagf0YaRkvQroAYhduXHjBmzatElQ4KCylABjTuOFhpbagivLvrzvtgwC+L6LQv0vOwUIwLLTmiwZUIAAJCxEVYAAFFV+Mk4AEgOiKkAAiio/GScAiQFRFSAARZWfjBOAxICoChCAospPxglAYkBUBQhAUeUn4wQgMSCqAgSgqPKTcQKQGBBVAQJQVPnJOAFIDIiqAAEoqvxknAAkBkRVgAAUVX4yTgASA6IqQACKKj8ZJwCJAVEVIABFlZ+ME4DEgKgKEICiyk/GCUBiQFQFCEBR5SfjBCAxIKoCBKCo8pNxApAYEFUBAlBU+ck4AUgMiKoAASiq/GScACQGRFWAABRVfjJOABIDoipAAIoqPxknAIkBURUgAEWVn4wTgMSAqAoQgKLKT8YJQGJAVAUIQFHlJ+MEIDEgqgIEoKjyk3ECkBgQVQECUFT5yTgBSAyIqgABKKr8ZPz/KYmDAlQjMS0AAAAASUVORK5CYII=</thumbnail><stage name="Stage" guid="541caacc-4c0e-4d35-b39a-9b0babafceb9" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite" idx="1" x="0" y="0" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><variables></variables><blocks></blocks><scripts><script x="716.9999999999999" y="335"><block id="5240" s="doIf"><block id="5241" s="reportEquals"><block id="5242" s="getLastAnswer"></block><block id="5243" var="random"/></block><script><block id="5244" s="doSayFor"><l>Correct!</l><l>2</l></block></script></block><block id="5220" s="doIf"><block id="5221" s="reportLessThan"><block id="5222" s="getLastAnswer"></block><block id="5223" var="random"/></block><script><block id="5224" s="doSayFor"><l>Too low!</l><l>2</l></block></script></block><block id="5228" s="doIf"><block id="5229" s="reportGreaterThan"><block id="5230" s="getLastAnswer"></block><block id="5231" var="random"/></block><script><block id="5232" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block></script><script x="425.0769230769231" y="162.0769230769231"><block id="5149" s="doDeclareVariables"><list><l>guess</l></list></block></script><script x="84.30769230769232" y="58.92307692307694"><block id="5077" s="receiveGo"></block><block id="5095" s="doSayFor"><l>Welcome to the guessing game!</l><l>2</l></block><block id="5115" s="doAsk"><l>What&apos;s your name?</l></block><block id="5096" s="doSayFor"><block id="5181" s="reportJoinWords"><list><l>Hello </l><block id="5196" s="getLastAnswer"></block></list></block><l>2</l></block><block id="5460" s="doAsk"><l>What is the minimum?</l></block><block id="5462" s="doSetVar"><l>min</l><block id="5464" s="getLastAnswer"></block></block><block id="5461" s="doAsk"><l>What is the maximum?</l></block><block id="5463" s="doSetVar"><l>max</l><block id="5465" s="getLastAnswer"></block></block><block id="5151" s="doSetVar"><l>random</l><block id="5180" s="reportRandom"><block id="5466" var="min"/><block id="5467" var="max"/></block></block><block id="5559" s="doSetVar"><l>guesses</l><block id="5560" s="reportNewList"><list></list></block></block><block id="5197" s="doUntil"><block id="5198" s="reportEquals"><block id="5199" s="getLastAnswer"></block><block id="5200" var="random"/></block><script><block id="5209" s="doAsk"><l>Guess a number</l></block><block id="5523" s="doUntil"><block id="5601" s="reportNot"><block id="5552" s="reportListContainsItem"><block id="5555" var="guesses"/><block id="5554" s="getLastAnswer"></block></block></block><script><block id="5620" s="doSayFor"><block id="5621" s="reportJoinWords"><list><l>You&apos;ve already guessed </l><block id="5622" s="getLastAnswer"></block></list></block><l>2</l></block><block id="5602" s="doAsk"><l>Guess a number</l></block></script></block><block id="5489" s="doAddToList"><block id="5557" s="getLastAnswer"></block><block id="5556" var="guesses"/></block><block id="5201" s="doIfElse"><block id="5202" s="reportEquals"><block id="5203" s="getLastAnswer"></block><block id="5204" var="random"/></block><script><block id="5205" s="doSayFor"><l>Correct!</l><l>2</l></block></script><script><block id="5212" s="doIfElse"><block id="5213" s="reportLessThan"><block id="5214" s="getLastAnswer"></block><block id="5215" var="random"/></block><script><block id="5216" s="doSayFor"><l>Too low!</l><l>2</l></block></script><script><block id="5206" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block></script></block></script></block></script><script x="453.53846153846155" y="322.6923076923078"><block id="5182" s="doUntil"><block id="5184" s="reportEquals"><block id="5186" s="getLastAnswer"></block><block id="5185" var="random"/></block><script><block id="5183" s="doIfElse"><block id="5187" s="reportLessThan"><block id="5188" s="getLastAnswer"></block><block id="5189" var="random"/></block><script><block id="5193" s="doSayFor"><l>Too low!</l><l>2</l></block></script><script><block id="5194" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block><block id="5192" s="doAsk"><l>Guess a number</l></block></script></block><block id="5195" s="doSayFor"><l>Correct!</l><l>2</l></block></script></scripts></sprite><watcher var="min" style="normal" x="10" y="36.600002" color="243,118,29"/><watcher var="max" style="normal" x="10" y="63.20000399999999" color="243,118,29"/><watcher var="guesses" style="normal" x="10" y="89.80000599999998" color="243,118,29" extX="104" extY="91"/><watcher var="random" style="normal" x="10" y="10" color="243,118,29"/></sprites></stage><hidden></hidden><headers></headers><code></code><blocks></blocks><variables><variable name="min"><l>1</l></variable><variable name="max"><l>20</l></variable><variable name="guesses"><list id="189"><item><l>10</l></item><item><l>3</l></item></list></variable><variable name="random"><l>5</l></variable></variables><editing></editing></project>