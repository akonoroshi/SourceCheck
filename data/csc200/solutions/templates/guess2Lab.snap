snapshot {
    @defBlock SAY {
        doSayFor(@anything)
    }
    @defBlock ASK {
        doAsk(@anything)
    }
    @defVar (GUESS_VAR) {
        stage {
            sprite {
                script {
                    receiveGo
                    @+defVar (SET_VAR, LAST, FIRST) {
                        @if (SET_VAR, FIRST) { doSetVar(varMenu:guesses, reportNewList(list)) }
                        @+repeat(1, 2) {
                            @block SAY
                        }
                        @block ASK
                        doSayFor(reportJoinWords(list(literal, getLastAnswer)), literal)
                        doAsk
                        doSetVar(var:min, getLastAnswer)
                        doAsk
                        doSetVar(var:max, getLastAnswer)
                        doSetVar(var:secret, reportRandom(var:min, var:max))
                        @if (SET_VAR, LAST) { doSetVar(varMenu:guesses, reportNewList(list)) }
                    }
                    @defBlock GUESS {
                        @if (GUESS_VAR) { var:guess }
                        @unless (GUESS_VAR) { getLastAnswer }
                    }
                    @defBlock GET_GUESS {
                        @block ASK
                        doUntil(reportNot(reportListContainsItem(var:guesses, getLastAnswer)), script {
                            @+optional {
                                doSayFor(@anything, literal)
                            }
                            doAsk(literal)
                        })
                        doAddToList(getLastAnswer, var:guesses)
                        @if (GUESS_VAR) { doSetVar(varMenu:guess, getLastAnswer) }
                    }
                    @defBlock EQ_COMPARE {
                        reportEquals(@block GUESS, var:secret)
                    }
                    @defBlock NOT_EQ_COMPARE {
                        reportLessThan(@or {
                            @inline { @block GUESS, var:secret }
                            @inline { var:secret, @block GUESS }
                        })
                    }

                    @defVar (ASK_TOP) {
                        @defVar (CORRECT_IN_LOOP) {
                            @defBlock EQ_FEEDBACK {
                                @if (CORRECT_IN_LOOP) {
                                    doIf(@block EQ_COMPARE, script {
                                        @block SAY
                                    })
                                }
                            }
                            @defBlock NOT_EQ_FEEDBACK_IFS {
                                @anyOrder {
                                    doIf(reportLessThan(@block GUESS, var:secret), script {
                                        @block SAY
                                        @if (ASK_TOP) { @block GET_GUESS }
                                    })
                                    doIf(reportLessThan(var:secret, @block GUESS), script {
                                        @block SAY
                                        @if (ASK_TOP) { @block GET_GUESS }
                                    })
                                }
                            }
                            @defBlock NOT_EQ_FEEDBACK_IF_ELSE {
                                doIfElse(@block NOT_EQ_COMPARE, script {
                                    @block SAY
                                }, script {
                                    @block SAY
                                })
                                @if (ASK_TOP) { @block GET_GUESS }
                            }
                            @defBlock NOT_EQ_FEEDBACK {
                                @or {
                                    @block NOT_EQ_FEEDBACK_IFS
                                    @block NOT_EQ_FEEDBACK_IF_ELSE
                                }
                            }

                            @if (ASK_TOP) { @block GET_GUESS }
                            doUntil(@block EQ_COMPARE, script {
                                @unless (ASK_TOP) { @block GET_GUESS }
                                @or {
                                    @inline {
                                        @block EQ_FEEDBACK
                                        @if (ASK_TOP) { @block NOT_EQ_FEEDBACK }
                                        @unless (ASK_TOP) { @block NOT_EQ_FEEDBACK_IFS }
                                    }
                                    doIfElse(@block EQ_COMPARE, script {
                                        @if (CORRECT_IN_LOOP) { @block SAY }
                                    }, script {
                                        @block NOT_EQ_FEEDBACK
                                    })
                                }
                            })
                            @unless (CORRECT_IN_LOOP) { @block SAY }
                        }
                    }

                }
            }
        }
        @anyOrder {
            varDec:secret
            @if (GUESS_VAR) { varDec:guess }
            varDec:min
            varDec:max
            varDec:guesses
        }
    }
}